packages/makeup-modal/src/index.js

```diff
@@ -2,13 +2,67 @@
 
 const keyboardTrap = require('makeup-keyboard-trap');
 const screenreaderTrap = require('makeup-screenreader-trap');
+const defaultOptions = { hoist: false };
+
+let hoistEl;
+let hoistedElementPlaceholder;
+let inertContentEl;
+let bodyChildIndexes = [];
+
+function unhoist() {
+    if (hoistEl) {
+        [...inertContentEl.childNodes].forEach((child) => {
+            if (!child.src) {
+                const index = bodyChildIndexes.shift();
+                if (index > document.body.childNodes.length) {
+                    document.body.appendChild(child);
+                } else {
+                    document.body.insertBefore(child, document.body.childNodes[index + 1]);
+                }
+            }
+        });
+        inertContentEl.remove();
+        inertContentEl = null;
+        bodyChildIndexes = [];
+
+        if (hoistedElementPlaceholder) {
+            hoistedElementPlaceholder.replaceWith(hoistEl);
+            hoistedElementPlaceholder = null;
+        }
+        hoistEl = null;
+    }
+    return hoistEl;
+}
+
+function hoist(el) {
+    unhoist();
+    hoistEl = el;
+
+    hoistedElementPlaceholder = document.createElement('div');
+    hoistEl.parentElement.insertBefore(hoistedElementPlaceholder, hoistEl);
+
+    inertContentEl = document.createElement('div');
+    [...document.body.childNodes].forEach((child, index) => {
+        if (!child.src) {
```

Comment:
Use `child.tagName.toLowerCase() === 'script'`. We might need to check for stylesheets too.